tests/Spec/Nri/Ui/SegmentedControl.elm

```diff
@@ -0,0 +1,248 @@
+        \() ->
+            program
+                |> ensureOnlyOneTabInSequence [ "Control 0", "Control 1", "Control 2" ]
+                |> done
+    , test "moves focus right on right arrow key" <|
+        \() ->
+            program
+                |> ensureTabbable "Control 0"
+                |> releaseRightArrow
+                |> ensureTabbable "Control 1"
+                |> ensureOnlyOneTabInSequence [ "Control 0", "Control 1", "Control 2" ]
+                |> releaseRightArrow
+                |> ensureTabbable "Control 2"
+                |> done
+    , test "moves focus left on left arrow key" <|
+        \() ->
+            program
+                |> ensureTabbable "Control 0"
+                |> releaseRightArrow
+                |> ensureTabbable "Control 1"
+                |> releaseLeftArrow
+                |> ensureTabbable "Control 0"
+                |> ensureOnlyOneTabInSequence [ "Control 0", "Control 1", "Control 2" ]
+                |> done
+    , test "when the focus is on the first element, move focus to the last element on left arrow key" <|
+        \() ->
+            program
+                |> ensureTabbable "Control 0"
+                |> releaseLeftArrow
+                |> ensureTabbable "Control 2"
+                |> ensureOnlyOneTabInSequence [ "Control 0", "Control 1", "Control 2" ]
+                |> done
+    , test "when the focus is on the last element, move focus to the first element on right arrow key" <|
+        \() ->
+            program
+                |> ensureTabbable "Control 0"
+                |> releaseLeftArrow
+                |> ensureTabbable "Control 2"
+                |> releaseRightArrow
+                |> ensureTabbable "Control 0"
+                |> ensureOnlyOneTabInSequence [ "Control 0", "Control 1", "Control 2" ]
+                |> done
+    ]
+
+
+ensureTabbable : String -> TestContext -> TestContext
+ensureTabbable word testContext =
+    testContext
+        |> ensureView
+            (Query.find [ Selector.attribute Role.tab, Selector.attribute (Key.tabbable True) ]
+                >> Query.has [ Selector.text word ]
+            )
+
+
+ensurePanelsFocusable : List String -> TestContext -> TestContext
+ensurePanelsFocusable words testContext =
+    testContext
+        |> ensureView
+            (Query.findAll [ Selector.attribute Role.tabPanel, Selector.attribute (Key.tabbable True) ]
+                >> Expect.all (List.indexedMap (\i w -> Query.index i >> Query.has [ Selector.text w ]) words)
+            )
+
+
+ensurePanelDisplayed : String -> TestContext -> TestContext
+ensurePanelDisplayed word testContext =
+    testContext
+        |> ensureView
+            (Query.find [ Selector.attribute Role.tabPanel, Selector.style "display" "block" ]
+                >> Query.has [ Selector.text word ]
+            )
+
+
+ensureOnlyOnePanelDisplayed : List String -> TestContext -> TestContext
+ensureOnlyOnePanelDisplayed panels testContext =
```

Comment:
I wonder if we could adjust this helper so that it asserts that the specific panels that we pass in are display:none with one panel selected?

Maybe:
```
ensureOnlyOnePanelDisplayed : String -> List String -> TestContext -> TestContext
ensureOnlyOnePanelDisplayed selectedPanel panels testContext =
```

? 

Otherwise, it seems a little odd to me to pass in a list of strings, when what we're using the list for is just the length, you know?